### PR TITLE
btf: prepare for making the package public

### PR DESCRIPTION
**Pull request by lmb**
I think this is the MVP to make `btf` accessible to users.

### CODE REVIEW

File: internal/btf/btf.go [hunk 1]
```diff
@@ -593,43 +581,8 @@ func (h *Handle) FD() int {
 
 // Map is the BTF for a map.
 type Map struct {
```
Comment (lmb) [inline internal/btf/btf.go:583]: * We could move this type to `ebpf` and hide the fields instead. What would we call the type? `ebpf.MapBTF`?
* We could embed these fields in `MapSpec` directly.

Comment (brb) [reply internal/btf/btf.go:583]: The first option sounds right to me. The type doesn't seem to have much to do with other types in the `btf` package. `ebpf.MapBTF` :+1:

**Review by brb — APPROVED** [review]
Thanks for working on this.

File: internal/btf/btf.go (hunk 1 above)
Comment (ti-mo) [reply internal/btf/btf.go:583]: Why would `btf.Map` be moved and `btf.Program` be kept in place?

Comment (lmb) [reply internal/btf/btf.go:583]: I'd say we keep it as it is now then?

**Comment by lmb** [comment]
@brb I pushed 4f00d3d3333b911474142732ece1e8dac29a7e52 which changes the type copy behaviour we talked about.

@ti-mo are you OK with proceeding with this?

**Pull request merged**